File: DollarCoin/flaskDollarApp/Nodejs/bot.js
const TelegramBot = require('node-telegram-bot-api');
const fs = require('fs');
const token = '7424542157:AAE0-u3H_tc4ReJ1Ju9Zij9GrZ-FW08lv2A';
const bot = new TelegramBot(token, { polling: true });

const USERS_FILE = './users.json';
let users = fs.existsSync(USERS_FILE) ? JSON.parse(fs.readFileSync(USERS_FILE)) : {};

function saveUsers() {
  fs.writeFileSync(USERS_FILE, JSON.stringify(users, null, 2));
}

bot.onText(/\/start(?:\s+(\d+))?/, (msg, match) => {
  const chatId = msg.chat.id;
  const userId = msg.from.id.toString();
  const referrerId = match[1];

  if (!users[userId]) {
    users[userId] = {
      id: userId,
      name: msg.from.first_name,
      referredBy: referrerId || null,
      balance: 0
    };
    saveUsers();
    console.log(`✅ New user added: ${userId}`);
  }

  bot.sendMessage(chatId, `👋 Welcome ${msg.from.first_name}!`, {
    reply_markup: {
      inline_keyboard: [
        [{ text: "⛏️ Start Mining", callback_data: "mine" }],
        [{ text: "💰 Check Balance", callback_data: "balance" }]
      ]
    }
  });
});

bot.on('callback_query', (query) => {
  const chatId = query.message.chat.id;
  const userId = query.from.id.toString();
  const action = query.data;

  if (action === 'mine') {
    users[userId].balance += 1;
    saveUsers();
    bot.sendMessage(chatId, `⛏️ Mining... You earned 1 coin!`);
  } else if (action === 'balance') {
    const balance = users[userId].balance;
    bot.sendMessage(chatId, `💰 Your balance is: ${balance} coins`);
  }
});

File: DollarCoin/flaskDollarApp/bot.py
from telegram import WebAppInfo, InlineKeyboardButton, InlineKeyboardMarkup
from telegram.ext import Application, CommandHandler

print('/start clicked')


async def start(update, context):
    image_path = 'coin.png'
    user = update.effective_user
    username = update.message.from_user.username
    chat = update.message.chat_id
    print(chat)
    print(user.first_name)
    message_text = update.message.text or ""
    parts = message_text.split()
    referral_code = parts[1] if len(parts) > 1 else None
    print(f"Received the referral code: {referral_code}")

    # Create an InlineKeyboardButton with a URL
    play_button = WebAppInfo(url=f"https://dollar.pythonanywhere.com")
    telegram_button = InlineKeyboardButton("Join our community", url="https://t.me/dollar_coin_channel")
    twitter_button = InlineKeyboardButton("follow us on X", url="https://x.com/Dollarcoin32536?t=Gt4UETs7z0mmJppG6-rVLQ&s=0")

    # Create an InlineKeyboardMarkup with the button
    reply_markup = InlineKeyboardMarkup([[InlineKeyboardButton("Play 🎮", web_app=play_button)], [telegram_button], [twitter_button]])



    # Send a message with the button
    with open(image_path, 'rb') as image_file:
        await update.message.reply_photo(photo=image_file, caption=f'Hi @{username}, tap on the coin and see your balance rise. DollarCoin is a decentralized exchange on the blockchain.', reply_markup=reply_markup)

# Initialize the Application
application = Application.builder().token("7424542157:AAE0-u3H_tc4ReJ1Ju9Zij9GrZ-FW08lv2A").build()

# Add a handler for the /start command
application.add_handler(CommandHandler("start", start))

# Start the bot
application.run_polling()
